### PR TITLE
Fixes again

### PR DESCRIPTION
**Pull request by Nekotekina**
Thanks to @Farseer2 for debugging cond_variable timeout @ #3708

### CODE REVIEW

**Comment by of1rsa** [comment]
Isn't it a duplicate of #3708 ?

**Comment by AniLeo** [comment]
This one is a proper fix

**Comment by Farseer2** [comment]
My fix is the same, no need to shame it - and either way at least credit it, its an open source project, its volunteering.

**Comment by AniLeo** [comment]
Updated pull request to add credits

**Comment by Nekotekina** [comment]
@Farseer2 I'm not shaming anything, you did a great job to find the bug. But writing a complete fix is a different problem.

**Comment by Megamouse** [comment]
What are the exact benefits of this solution compared to the other?
Why exactly is this solution preferred?
Maybe if you provide a quick explanation it might help us understand.

**Comment by Farseer2** [comment]
@Nekotekina Meant Ani calling the fix 'not proper'. And this PR is the exact same fix, just with a variable instead of a magic (-1), which already existed. 
But I digress, the point is this is what Github comments are for, and this is wrong, especially as commits like this are getting people to pay your Patreon - I just see this as a wrong this to do on your side.

**Comment by Nekotekina** [comment]
Then you didn't really read the code I wrote. Consider the time it would took me to ask you to do the same changes via comments.

**Comment by Farseer2** [comment]
Considering the fact you get paid, yeah, I guess you should comment instead, as its originally someone else's work, not yours.
It isn't the first nor second time I've seen it happen here, and it isn't this specific PR that bothers me, but the principal behind it, the reason I mentioned it now is of1rsa's comment.
And as for this case, it also isn't as if merging this PR was crucial. It is rather minor, and there are lots of PRs sitting for weeks and months without getting merged - this PR itself was open for over 18 hours until getting merged, writing "check for overflow in Linux as well" doesn't take that long. 

The bottom line is that I honestly think if someone went through the trouble of debugging, testing, finding the bug, fixing it and creating a PR all in their free time - rewriting it and pushing it to master under your name is disrespectful both towards the programmer and towards yourself and your patreons, not specifically in the context of this PR, but generally speaking.

**Comment by Nekotekina** [comment]
OK, would it make a difference if I modified your commit and pushed that it under your name instead? A lot of time is wasted on communications sometimes, not on writing code.

**Comment by Farseer2** [comment]
Its fine, no need to play git games for that, this really is a minor commit - it was the concept that bothered me. If you do want to edit someone's PR though, pretty sure that in Github when someone PRs to your repository you get write permissions to their source branch (by default), so you can leave stuff there - might be wrong, though.

Anyway, sorry, didn't mean to start a discussion here, just wanted to point out to you something that bothered me and at least 1 other person that I know of.

**Comment by Megamouse** [comment]
Please just make a commit or better a PR on their branch in the future. Then merge both commits to have the initial commit in github. Problem solved. No need for funky counter PRs or (be it legit or not) drama like this.

And it's not like we're on a schedule here, so why all the haste Neko.

I get that it's easy to push your own fix but it's not the most sensitive solution as we've seen today.

Can we all be cool now? Thx
